shared/constants/types/teams.tsx

```diff
@@ -151,7 +156,8 @@ export type State = Readonly<{
   teamnames: Set<Teamname> // TODO remove
   teammembercounts: I.Map<Teamname, number>
   teamProfileAddList: Array<TeamProfileAddList>
-  newTeams: Set<string>
-  newTeamRequests: Array<string>
+  newTeams: Set<TeamID>
+  newTeamRequests: Map<TeamID, number>
+  newTeamRequestsByName: Map<string, number> // TODO remove
```

Comment:
Opted to make this temporary field to support parts of the view that haven't been updated instead of a larger protocol change to include both the name and ID here.